client/blocks/login/index.jsx

```diff
@@ -168,12 +169,18 @@ class Login extends Component {
 			translate,
 			twoStepNonce,
 			fromSite,
+			currentUser,
+			redirectTo,
 		} = this.props;
 
 		let headerText = translate( 'Log in to your account' );
 		let preHeader = null;
 		let postHeader = null;
 
+		if ( currentUser ) {
+			postHeader = <ContinueAsUser user={ currentUser } redirectUrl={ redirectTo || '/' } />;
```

Comment:
@jsnajdr : I'm not confident this is the correct way to handle the redirect URL. As far as I can tell, `getRedirectToSanitized` only gets populated **after** a successful login attempt. Is there a better way of getting a sanitised redirect URL here?

Reply:
Yes, apparently the validation is server-side and part of the login endpoint. We'll need a separate redirect validation endpoint to make this feature work.

Some pointers I found:
- the server-side validation is done in `Login_Base_Endpoint::create_response_with_redirect_url` method in the wpcom repo.
- was implemented by @stephanethomas and @Tug in D5959-code
- #14872 has interesting historical info how we used to do the validation client-side

Possible solution: new endpoint:
```
/rest/v1.1/users/$user/validate-redirect
```
(modeled after existing `/users/$user/auth-options` endpoint queried after you enter username in login)

One possible caveat: the `login_redirect` and `logout_redirect` filters take `$user` as a parameter, but here they would be called anonymously, before login, being exposed to public. Is it safe to call them in that case? Can they be used to leak info about the user?

@josephscott might help a lot here, knowing about all flavors of WordPress login 🙂

Reply:
I'd suggest any redirect code get some help from the security folks.  We want to be careful not to send people some where outside of our control.

Reply:
> One possible caveat: the `login_redirect` and `logout_redirect` filters take `$user` as a parameter, but here they would be called anonymously, before login, being exposed to public.

I realized this is not true: the `/users/$user/validate-redirect` endpoint can be authenticated only. Calypso Login page first realizes that the user is already logged in, and only then offers "continue as user" and verifies the redirect link.

Reply:
Thanks, @jsnajdr and @josephscott! Any suggestions of where I should go for help in making this API endpoint? My PHP-fu is weak 😞

Reply:
I'll try to help with the endpoint.

Reply:
Thank yo so much, @jsnajdr! 🙇

Reply:
Added new endpoint in D31621:
```
/rest/v1.1/me/validate-redirect?redirect_url=...
```
@sgomes With that patch applied, you can continue developing this feature.

Reply:
Thank you so much for this again, @jsnajdr 🙏 You rock! ❤️